src/coq_rules.ml

```diff
@@ -173,21 +174,36 @@ let setup_rules ~sctx ~dir ~dir_contents (s : Dune_file.Coq.t) =
             ~ml_iflags ~mlpack_rule) coq_modules in
   coq_rules
 
+(* This is here for compatibility with Coq < 8.11, which expects the
+   plugin's `.cmxs` files to be in the folder containing the `.vo`
+   files *)
+let coq_cmx_install_rules ~dst_dir libs =
+  let rules_for_lib lib =
+    Mode.Dict.get (Lib.plugins lib) Mode.Native |>
+    List.map ~f:(fun plugin_file ->
+      let dst = Path.(to_string (relative dst_dir (basename plugin_file))) in
+      None, Install.(Entry.make Section.Lib_root ~dst plugin_file))
+  in
+  List.concat_map ~f:rules_for_lib libs
+
 let install_rules ~sctx ~dir s =
   match s with
   | { Dune_file.Coq. public = None; _ } ->
     []
-  | { Dune_file.Coq. public = Some { package = _ ; _ } ; _ } ->
+  | { Dune_file.Coq. public = Some { package = _ ; _ } ; loc; _ } ->
+    let scope = SC.find_scope_by_dir sctx dir in
     let dir_contents = Dir_contents.get_without_rules sctx ~dir in
     let name = Dune_file.Coq.best_name s in
+    (* This is the usual root for now, Coq + Dune will change it! *)
+    let coq_root = "coq/user-contrib" in
```

Comment:
You should probably turn it into a path here.

Reply:
Fixed, also incorporated the removal of a local open you did in other PR.